beta-src/src/hooks/useViewport.ts

```diff
@@ -20,5 +20,10 @@ export default function useViewport(): [Viewport, () => void] {
     setValue(getViewport());
   };
 
+  React.useEffect(() => {
+    window.addEventListener("resize", setViewport);
+    return () => window.removeEventListener("resize", setViewport);
+  }, []);
```

Comment:
the problem i see with this here is each time we use this, we add an event listener. as discussed earlier, we should put the useViewport logic WDMain and make the getDevice logic dispatch the device stuff to a redux slice that is separate from game-slice. anything that needs to know the device can useAppSelector hook to get the device information. we should modify getDevice to do the logic for isMobileLandscape and have that in state too.

Reply:
Okay, let's push what we have now and tomorrow I'll focus on this as a part of tech debt since we've spent a lot of time on this on Friday already and the priority is UAT.